src/app/actions.ts

```diff
@@ -3,33 +3,29 @@
 import { redirect } from "next/navigation";
 import { createRegistration } from "@/lib/registration/create-registration";
 import { registrationSchema } from "@/lib/schemas";
+import { AgentRegistrationFormData } from "@/app/types";
 
 export async function registerAgent(formData: FormData) {
   // Extract all form fields and create nested structure
-  const rawFormData = {
+  const rawFormData: AgentRegistrationFormData = {
```

Comment:
It is intentional to not annotate this object. At this point, we do not have a properly formed schema data structure since we should not implicitly trust what we get from the client. 

Let's remove the assertion to have the zod schema properly do this.